lib/post/widgets/comment_view.dart

```diff
@@ -96,19 +103,42 @@ class _CommentSubviewState extends State<CommentSubview> with SingleTickerProvid
       }
     });
 
-    if (widget.selectedCommentId != null) {
-      WidgetsBinding.instance.addPostFrameCallback((_) {
-        // If we are looking at a comment context, scroll to the first comment.
-        // The delay is purely for aesthetics and is not required for the logic to work.
-        Future.delayed(const Duration(milliseconds: 250), () {
-          widget.itemScrollController.scrollTo(index: 1, duration: const Duration(milliseconds: 250));
-        });
+    // The following logic helps us to set the size of the bottom spacer so that the user can scroll the last comment
+    // to the top of the viewport but no further.
+    // This must be run some time after the layout has been rendered so we can measure everything.
+    // It also must be run after there is something to scroll, and the easiest way to do this is to do it in a scroll listener.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      widget.itemScrollController.primaryScrollController?.addListener(() {
+        double lastCommentTopOffset = widget.itemPositionsListener.itemPositions.value.toList().reversed.skip(2).first.itemLeadingEdge;
```

Comment:
I looked through this and I believe we can remove this `lastCommentTopOffset` logic altogether, and just simply add `_lastCommentKey.currentContext != null)` to the conditional statement below and it should still work?

e.g.,

```dart
if (_bottomSpacerHeight == null && lastCommentTopOffset > 0) {
```

becomes 
```dart
if (_bottomSpacerHeight == null && _lastCommentKey.currentContext != null) {
```

This would remove our dependency on `itemPositionsListener` and allow us to merge this and #1361! Let me know if this works!

Reply:
I seem to recall there was an issue where this was getting calculated too soon, hence the check to ensure that the `lastCommentTopOffset` was a reasonable value (I recall seeing it negative, which was weird). But it seems to work fine the way you suggested, so let's go with it!

Reply:
I'll keep a lookout for any inconsistencies here!